packages/fxa-auth-server/lib/payments/iap/iap-config.ts

```diff
@@ -2,12 +2,30 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 import { Firestore } from '@google-cloud/firestore';
+import {
+  MozillaSubscriptionTypes,
+  SubscriptionType,
+} from 'fxa-shared/subscriptions/types';
 import { Container } from 'typedi';
 import { TypedCollectionReference } from 'typesafe-node-firestore';
 
 import { AppConfig, AuthFirestore, AuthLogger } from '../../types';
+import { SubscriptionPurchase as AppStoreSubscriptionPurchase } from './apple-app-store/subscription-purchase';
+import { SubscriptionPurchase as PlayStoreSubscriptionPurchase } from './google-play/subscription-purchase';
 import { IapConfig } from './types';
 
+export function getIapPurchaseType(
+  purchase: PlayStoreSubscriptionPurchase | AppStoreSubscriptionPurchase
+): Omit<SubscriptionType, typeof MozillaSubscriptionTypes.WEB> {
+  if (purchase.hasOwnProperty('purchaseToken')) {
+    return MozillaSubscriptionTypes.IAP_GOOGLE;
+  }
+  if (purchase.hasOwnProperty('originalTransactionId')) {
+    return MozillaSubscriptionTypes.IAP_APPLE;
+  }
```

Comment:
My previous implementation had this logic, but it was [permafailing CI](https://app.circleci.com/pipelines/github/mozilla/fxa/29853/workflows/b4836956-b198-45fb-98d4-c3fa62e9d112/jobs/329537) despite passing locally.
```js
 if (purchase instanceof PlayStoreSubscriptionPurchase) {
   return MozillaSubscriptionTypes.IAP_GOOGLE;
 }
 if (purchase instanceof AppStoreSubscriptionPurchase) {
   return MozillaSubscriptionTypes.IAP_APPLE;
 }
```
Would you have any idea why?

Reply:
Per Ben, this looks fine to him; try it again since a few things were changed since that CI run.

Reply:
Hm so one thing I noticed is that the `instanceof` logic is fine when I run the tests in isolation but it fails when I run the test as part of a suite (e.g. `yarn test test/local/`).